src/main/scala/com/microsoft/hyperspace/actions/RefreshActionBase.scala

```diff
@@ -71,14 +71,16 @@ private[actions] abstract class RefreshActionBase(
   // Reconstruct a df from schema
   protected lazy val df = {
     val relations = previousIndexLogEntry.relations
-    val latestRelation = Hyperspace
+    val relationMetadata = Hyperspace
       .getContext(spark)
       .sourceProviderManager
       .getRelationMetadata(relations.head)
-      .refresh()
+    val latestRelation = relationMetadata.refresh()
     val dataSchema = DataType.fromJson(latestRelation.dataSchemaJson).asInstanceOf[StructType]
-    val df = spark.read
-      .schema(dataSchema)
+    var df = {
```

Comment:
`val` + `}.format(...` in my previous comment is done by scalafmt

Reply:
Done